protocol/protocol.tex

```diff
@@ -6822,6 +7713,112 @@
 } %sapling
 
 
+\orchard{
+\introlist
+\lsubsubsubsection{Sinsemilla Hash Function}{concretesinsemillahash}
+
+\defining{$\SinsemillaHash$} is an algebraic \hashFunction with
+\collisionResistance (for fixed input length) derived from assumed hardness
+of the Discrete Logarithm Problem. It is designed by Sean Bowe and Daira Hopwood.
+The motivation for introducing a new discrete-log-based hash function (rather than
+using $\PedersenHash$) is to make efficient use of the lookups available in recent
+proof systems including \HaloTwo.
+
+$\SinsemillaHash$ is used in the definition of $\SinsemillaCommit{}$
+(\crossref{concretesinsemillacommit}), and for the \Orchard{} \incrementalMerkleTree
+(\crossref{orchardmerklecrh}).
+
+Let $\GroupP$, $\ZeroP$, $\ParamP{q}$, $\ParamP{r}$, and $\ParamP{b}$ be as defined in
+\crossref{pallasandvesta}.
+
+Let $\ExtractP \typecolon \GroupP \rightarrow \MerkleHash{Orchard}$ be as
+defined in \crossref{concreteextractorpallas}.
+
+Let $\GroupPHash$ be as defined in \crossref{concretegrouphashpallasandvesta}.
+
+Let $\Uncommitted{Orchard}$ be as defined in \crossref{constants}.
+
+Let $\LEBStoOSP{} \typecolon (\ell \typecolon \Nat) \times \bitseq{\ell} \rightarrow \byteseq{\sceiling{\ell/8}}$
+and $\LEOStoIP{} \typecolon (\ell \typecolon \Nat \suchthat \ell \bmod 8 = 0) \times \byteseq{\ell/8} \rightarrow \binaryrange{\ell}$
+be as defined in \crossref{endian}.
+
+\vspace{1ex}
+Let $k = 11$.
+
+Let $c$ be the largest integer such that $2^n \leq \hfrac{\ParamP{r}-1}{2}$,
+i.e.\ $c := 253$.
+
+Define $\SinsemillaGenInit \typecolon \byteseqs \rightarrow \GroupPstar$ and
+$\SinsemillaGenBase \typecolon \binaryrange{k} \rightarrow \GroupPstar$ by:
+
+\begin{tabular}{@{\hskip 1.5em}r@{\;}l}
+  $\SinsemillaGenInit(D)$ &$:= \GroupPHash\!\big(\ascii{z.cash:SinsemillaQ}, D\big)$ \\
+  $\SinsemillaGenBase(j)$ &$:= \GroupPHash\!\big(\ascii{z.cash:SinsemillaS}, \LEBStoOSPOf{32}{\ItoLEBSPOf{32}{j}}\kern-0.25em\big)$.
```

Comment:
There's a type error here: `S(j)` takes `j` as an integer, but in `SinsemillaHashToPoint` we call it as `S(M_i)` where `M_i` is a bitstring.

Reply:
This should be `\SinsemillaGenBase(\LEBSPtoIP{k}(M_i))`.